src/go/rpk/pkg/cli/cmd/cluster/txn/find_hanging.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package txn
+
+import (
+	"sort"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"golang.org/x/exp/maps"
+)
+
+func newFindHangingCommand(fs afero.Fs) *cobra.Command {
+	var timeout time.Duration
+	cmd := &cobra.Command{
+		Use:   "find-hanging",
+		Short: "Find hanging transactions",
+		Long: `Find hanging transactions.
+
+This command attempts to find all transactions that are "hanging". A hanging
+transaction is one that has been open for a long time with no progress being
+made: no records are being produced, the transaction is just in the open state.
+The --max-transaction-timeout flag controls how long a transaction can be open
+with no progress before it is considered hanging. If no transactions are
+hanging, this will output a few headers with no rows.
+`,
+
+		Run: func(cmd *cobra.Command, txnIDs []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			producers, err := adm.DescribeProducers(cmd.Context(), nil)
+			out.HandleShardError("DescribeProducers", err)
+
+			// All producers that have a start offset and the last
+			// timestamp is older than we want are candidates for
+			// hanging.
+			now := time.Now()
+			candidates := make(map[int64][]kadm.DescribedProducer)
+			producers.EachProducer(func(p kadm.DescribedProducer) {
+				if p.CurrentTxnStartOffset == -1 {
+					return
+				}
+				if now.Sub(time.UnixMilli(p.LastTimestamp)) < timeout {
+					return
+				}
+				candidates[p.ProducerID] = append(candidates[p.ProducerID], p)
+			})
+
```

Comment:
Should we fail early here if `candidates` is empty?

Reply:
* Moving table headers up, adding an early return